### PR TITLE
Fixed it

### PR DESCRIPTION
**Pull request by martinthomson**

### CODE REVIEW

File: neqo-transport/src/connection/mod.rs [hunk 1]
```diff
@@ -1745,12 +1745,12 @@ impl Connection {
             // Make a path on which to run the handshake.
             self.setup_handshake_path(path, now);
 
-            self.zero_rtt_state = match self.crypto.enable_0rtt(self.version, self.role) {
-                Ok(true) => {
-                    qdebug!([self], "Accepted 0-RTT");
-                    ZeroRttState::AcceptedServer
-                }
-                _ => ZeroRttState::Rejected,
+            self.zero_rtt_state = if self.crypto.enable_0rtt(self.version, self.role) == Ok(true) {
```
Comment (martinthomson) [inline neqo-transport/src/connection/mod.rs:1748]: This is just factoring and more logging.

File: neqo-transport/src/connection/tests/zerortt.rs [hunk 2]
```diff
@@ -277,17 +277,15 @@ fn zero_rtt_loss_accepted() {
         let mut now = now();
         let client_stream_id = client.stream_create(StreamType::UniDi).unwrap();
         client.stream_send(client_stream_id, &[1, 2, 3]).unwrap();
-        let ci = client.process(None, now);
+        let mut ci = client.process(None, now);
```
Comment (martinthomson) [inline neqo-transport/src/connection/tests/zerortt.rs:280]: I removed a layer of nesting here and ensured that you passed in the correct packet (the last, not the first).

File: test-fixture/src/lib.rs [hunk 3]
```diff
@@ -53,7 +53,7 @@ pub fn fixture_init() {
 
 // This needs to be > 2ms to avoid it being rounded to zero.
 // NSS operates in milliseconds and halves any value it is provided.
-pub const ANTI_REPLAY_WINDOW: Duration = Duration::from_millis(10);
+pub const ANTI_REPLAY_WINDOW: Duration = Duration::from_millis(10000);
```
Comment (martinthomson) [inline test-fixture/src/lib.rs:56]: This is the real fix.

NSS estimates the RTT.  There is a very small allowance for dropped packets and variations in network delay.  The test you wrote added a bunch of extra delay that looked - at least to NSS - like additional delays.

File: neqo-transport/src/connection/tests/zerortt.rs (hunk 2 above)
Comment (larseggert) [reply neqo-transport/src/connection/tests/zerortt.rs:280]: Thanks! I have this locally but didn't push yet.

**Review by larseggert — APPROVED** [review]

**Pull request merged**